packages/app-desktop/gui/Sidebar/hooks/useSidebarListData.ts

```diff
@@ -0,0 +1,101 @@
+import { useMemo } from 'react';
+import { FolderListItem, HeaderId, HeaderListItem, ListItem, ListItemType, TagListItem } from '../types';
+import { FolderEntity, TagsWithNoteCountEntity } from '@joplin/lib/services/database/types';
+import { renderFolders, renderTags } from '@joplin/lib/components/shared/side-menu-shared';
+import { _ } from '@joplin/lib/locale';
+import CommandService from '@joplin/lib/services/CommandService';
+import Setting from '@joplin/lib/models/Setting';
+
+interface Props {
+	tags: TagsWithNoteCountEntity[];
+	folders: FolderEntity[];
+	collapsedFolderIds: string[];
+	folderHeaderIsExpanded: boolean;
+	tagHeaderIsExpanded: boolean;
+}
+
+const onAddFolderButtonClick = () => {
+	void CommandService.instance().execute('newFolder');
+};
+
+const onHeaderClick = (headerId: HeaderId) => {
+	const settingKey = headerId === HeaderId.TagHeader ? 'tagHeaderIsExpanded' : 'folderHeaderIsExpanded';
+	const current = Setting.value(settingKey);
+	Setting.setValue(settingKey, !current);
+};
+
+const useSidebarListData = (props: Props): ListItem[] => {
+	const tagItems = useMemo(() => {
+		return renderTags<ListItem>(props.tags, (tag): TagListItem => {
+			return {
+				kind: ListItemType.Tag,
+				tag,
+				key: tag.id,
+			};
+		});
+	}, [props.tags]);
+
+
+	const folderItems = useMemo(() => {
+		const renderProps = {
+			folders: props.folders,
+			collapsedFolderIds: props.collapsedFolderIds,
+		};
+		return renderFolders<ListItem>(renderProps, (folder, hasChildren, depth): FolderListItem => {
+			return {
+				kind: ListItemType.Folder,
+				folder,
+				hasChildren,
+				depth,
+				key: folder.id,
+			};
+		});
+	}, [props.folders, props.collapsedFolderIds]);
+
+	return useMemo(() => {
+		const foldersHeader: HeaderListItem = {
+			kind: ListItemType.Header,
+			label: _('Notebooks'),
+			iconName: 'icon-notebooks',
+			id: HeaderId.FolderHeader,
+			key: HeaderId.FolderHeader,
+			onClick: onHeaderClick,
+			onPlusButtonClick: onAddFolderButtonClick,
+			extraProps: {
+				['data-folder-id']: '',
+				toggleblock: '1',
```

Comment:
I'm not sure `toggleblock` is necessary — a `git grep toggleblock` produces:
```
git grep toggleblock                                                                                                                                                                                     ~/Documents/joplin
packages/app-desktop/gui/Sidebar/hooks/useSidebarListData.ts:                           toggleblock: '1',
packages/app-desktop/gui/Sidebar/hooks/useSidebarListData.ts:                           toggleblock: '1',
packages/tools/cspell/dictionary3.txt:toggleblock
```

I also don't see any [reference to it on MDN](https://developer.mozilla.org/en-US/search?q=toggleblock), so it doesn't seem to be a DOM API.

If removed, toggling sections by clicking on headers seems to work.

Reply:
It looks like usages of the `toggleblock` attribute were removed in 056285deda4f67b4e38a82b89145a3e52804a6ed ([when the Sidebar was migrated to TypeScript](https://github.com/laurent22/joplin/commit/056285deda4f67b4e38a82b89145a3e52804a6ed#diff-378103610d1c5ebbec952eb6f101765055dbe28fe76902fcb0b368365b012176L477)), so it should be safe to remove.